test/linux.rs

```diff
@@ -0,0 +1,71 @@
+cfg_if! {
+    if #[cfg(target_os = "linux")] {
+        mod linux_tests {
```

Comment:
Is this the right way to do platform independent integration tests?  At first I just added to the unit tests, but I couldn't get it to read files anywhere, so I went integration tests.